src/boson_sampling_utilities/boson_sampling_utilities.py

```diff
@@ -151,8 +151,20 @@ def calculate_number_of_possible_lossy_n_particle_m_mode_output_states(n: int, m
     return states_number
 
 
-def prepare_interferometer_matrix_in_expanded_space(interferometer_matrix: ndarray) -> ndarray:
+def get_modes_transmissivity_values_from_matrix(lossy_interferometer_matrix: ndarray) -> List[float]:
+    v_matrix, singular_values, u_matrix = svd(lossy_interferometer_matrix)
+    return [i ** 2 for i in singular_values]
+
+
+def prepare_interferometer_matrix_in_expanded_space_with_first_k_lossless_modes(interferometer_matrix: ndarray,
+                                                                                k: int) -> ndarray:
     v_matrix, singular_values, u_matrix = svd(interferometer_matrix)
+
+    if k > len(singular_values):
+        k = len(singular_values)
+    for i in range(k):
```

Comment:
Wouldn't `singular_values[:k] = 1` do the job?

Reply:
+1

Reply:
+1

Reply:
+1